Kconfig.zephyr

```diff
@@ -512,7 +512,14 @@ config BUILD_OUTPUT_META
 	  Create a build meta file in the build directory containing lists of:
 	  - Zephyr: path and revision (if git repo)
 	  - Zephyr modules: name, path, and revision (if git repo)
-	  - West projects: path and revision
+	  - West:
+	    - manifest: path and revision
+	    - projects: path and revision
+	  - Workspace:
+	    - dirty: one or more repositories are marked dirty
+	    - extra: extra Zephyr modules was manually included in the build
+	    - off:   the SHA of one or more west projects does not match the
+	             referenced by the west manifest (west only state)
```

Comment:
What does this mean? Does it mean that the HEAD commit is different than the manifest-rev commit? If so, I think we should just say that. If not, I think this needs to be made clearer; I can't understand what this is saying just from the Kconfig description.

Reply:
it is, because the head of the manifest-rev is what west returns as sha.
But are users of `west` familiar with the manifest-rev ?

What about:
```
- off:   the SHA of one or more west projects has diverged since the last `west update`
          and are therefore out of sync with the manifest (west only state).
```

Reply:
I think "west only state" is unnecessarily vague. That's why I said:

> Does it mean that the HEAD commit is different than the manifest-rev commit? If so, I think we should just say that.

I suggest:

```
- off:   the SHA of one or more west projects has diverged since the last `west update`
         (that is, the current commit is different than the manifest-rev commit)
```

Reply:
> I think "west only state" is unnecessarily vague. 

What that means is that this state is only present if `west` is used for workspace management.
I will not be present in non-west workspaces.

Reply:
>> I think "west only state" is unnecessarily vague.

> What that means is that this state is only present if west is used for workspace management.
> I will not be present in non-west workspaces.

Thanks for explaining, but that doesn't solve the problem with this pull request. My point is that this wording is vague and I have a suggestion for how to make it clearer. Writing a comment in the github pull request conversation clearing it up will not help people who are going to read the code later. What do you say to my suggestion?

Reply:
> My point is that this wording is vague and I have a suggestion for how to make it clearer.

Yes, I understand that.
And you're proposal is fine, but it lacks the part I also wanted to tell the reader, that this state can only be present if `west` is used, which were the point I tried to make with my last comment.

You may say that piece of information can be omitted, because it should be obvious when we refer to `west update` in the text, I just want to be sure you know the reason for that sentence so that it can be reworded, or we agree why it can be removed.

Reply:
(west only state) is not being explanatory unfortunately. It should either be omitted or re-written.

Reply:
same in the Kconfig help text

Reply:
> (west only state) is not being explanatory unfortunately. It should either be omitted or re-written.

I hope it's clear from earlier comments that I agree to this.
I just want to know whether omitting this or re-writing is preferred.

If there is consensus that this information is not relevant, then it's easier to remove.

Reply:
Whether it should be omitted or re-written depends on what you are trying to say.

You say that "the state is only present if west is used".

Does that mean that off : False, when west is not used, or that the attribute "off" is not present at all?

Reply:
I have the same question as @SebastianBoe .

Reply:
> Does that mean that off : False, when west is not used, or that the attribute "off" is not present at all?

Good point.
To make this clear, I have now ensured that `west` and statuses only related to `west` usage are now only present in the meta file when `west` is used.

So that means not present at all.

Reply:
Updated to consistent behavior, and description should be clearer now.

Thanks for the feedback.